lib/index.js

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const _constants = require('./constants');
+const bedrock = require('bedrock');
+const brJobs = require('bedrock-jobs');
+const cache = require('bedrock-redis');
+const {config, util: {clone}} = bedrock;
+const assert = require('assert-plus');
+const keys = require('./keys');
+
+require('./config');
+const cfg = config['stats-storage-redis'];
+
+bedrock.events.on('bedrock.start', async () => {
+  const jobQueue = exports._jobQueue = brJobs.addQueue(
+    {name: _constants.JOB_QUEUE});
+  // setup a processor for the queue with the default concurrency of 1
+  jobQueue.process(async () => {
+    const beforeDate = Date.now() - cfg.history.ttl;
+    return exports.trimHistory({beforeDate});
+  });
+  await jobQueue.add({}, {
+    // prevent duplicate jobs by specifying a non-unique jobId
+    jobId: 'trimHistory',
+    // repeated jobs are completed and rescheduled on every iteration
+    repeat: {every: cfg.history.trimInterval},
+    // do not keep record of successfully completed jobs in redis
+    removeOnComplete: true
+  });
+});
+
+/**
+ * @module bedrock-stats-storage-redis
+ */
+
+exports.keys = keys;
+
+/**
+ * Query stats history.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string[]} options.monitorIds - The monitorIds to query.
+ * @param {number|string} [options.endDate='+inf'] - The end date for the query
+ *   in ms since epoch.
+ * @param {number|string} [options.startDate='-inf'] - The start date for the
+ *   query in ms since epoch.
+ *
+ * @returns {Promise<Object>} A collated history for the specified monitorIds.
+ */
+exports.find = async ({endDate = '+inf', monitorIds, startDate = '-inf'}) => {
```

Comment:
Can we use the JavaScript symbols `Infinity` and `-Infinity` here instead? Do we want to use strings at this level? If we do want to use strings, is there a precedent for `+inf` or did we come up with that?

Reply:
This is from the redis API: https://redis.io/commands/zrangebyscore#exclusive-intervals-and-infinity

Reply:
So be it!